Sources/BraveWallet/Extensions/BraveWalletExtensions.swift

```diff
@@ -312,4 +312,9 @@ public extension String {
   var endsWithSupportedSNSExtension: Bool {
     WalletConstants.supportedSNSExtensions.contains(where: hasSuffix)
   }
+  
+  /// Returns true if the string ends with a supported UD extension.
+  var endsWithSupportedUDExtension: Bool {
+    WalletConstants.supportedUDExtensions.contains(where: hasSuffix)
```

Comment:
i think we missed in all ENS/SNS/UD that domain comparison should be case insensitive. if i type `jim-unstoppable.Crypto` it won't get resolved.

Reply:
In some discussion around this, just holding before merging for now.